download-basket/src/main/java/downloadbasket/helpers/SendDownloadDetailsToEmailThread.java

```diff
@@ -0,0 +1,296 @@
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.UUID;
+import java.util.Locale;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import downloadbasket.data.ErrorReportDetails;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceIbatisImpl;
+import fi.nls.oskari.util.PropertyUtil;
+import fi.nls.oskari.util.IOHelper;
+import downloadbasket.data.LoadZipDetails;
+import downloadbasket.data.NormalWayDownloads;
+import downloadbasket.data.ZipDownloadDetails;
+import org.apache.commons.mail.HtmlEmail;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ * Send download details email service (thread).
+ */
+public class SendDownloadDetailsToEmailThread implements Runnable {
+	JSONArray downLoadDetails;
+	JSONObject userDetails;
+	Locale locale;
+
+	private final Logger LOGGER = LogFactory.getLogger(SendDownloadDetailsToEmailThread.class);
+	private final String PARAM_CROPPING_MODE = "croppingMode";
+	private final String PARAM_CROPPING_LAYER = "croppingLayer";
+	private final String PARAM_LAYER = "layer";
+	private final String PARAM_LAYER_ID = "id";
+
+	/**
+	 * Constructor.
+	 *
+	 * @param downLoadDetails
+	 *            download details
+	 * @param userDetails
+	 *            user details
+	 */
+	public SendDownloadDetailsToEmailThread(JSONArray downLoadDetails, JSONObject userDetails, Locale locale) {
+		this.downLoadDetails = downLoadDetails;
+		this.userDetails = userDetails;
+		this.locale = locale;
+	}
+
+	private ArrayList<ZipDownloadDetails> downloadFromService() {
+		ArrayList<ZipDownloadDetails> mergeThese = new ArrayList<ZipDownloadDetails>();
+		final String strTempDir = PropertyUtil.get("oskari.wfs.download.folder.name");
+		NormalWayDownloads normalDownloads = new NormalWayDownloads();
+		for (String download : PropertyUtil.getCommaSeparatedList("oskari.wfs.download.normal.way.downloads")) {
+			normalDownloads.addDownload(download);
+		}
+		try {
+			DownloadServices ds = new DownloadServices();
+			for (int i = 0; i < downLoadDetails.length(); i++) {
+				JSONObject download = downLoadDetails.getJSONObject(i);
+				final String croppingMode = download.getString(PARAM_CROPPING_MODE);
+				String croppingLayer = "";
+				if (download.has(PARAM_CROPPING_LAYER)) {
+					croppingLayer = download.getString(PARAM_CROPPING_LAYER);
+				}
+
+				LoadZipDetails ldz = new LoadZipDetails();
+				ldz.setTemporaryDirectory(strTempDir);
+				ldz.setUserEmail(userDetails.getString("email"));
+				ldz.setLanguage(this.locale.getLanguage());
+				ldz.setDownloadNormalWay(normalDownloads.isBboxCropping(croppingMode, croppingLayer));
+				OskariLayerService mapLayerService = new OskariLayerServiceIbatisImpl();
```

Comment:
Shouldn't create multiple instances of `OskariLayerServiceIbatisImpl`, definetely not inside a loop